src/passes/SimplifyLocals.cpp

```diff
@@ -140,9 +140,9 @@ struct SimplifyLocals
     } else if (curr->is<If>()) {
       assert(!curr->cast<If>()
                 ->ifFalse); // if-elses are handled by doNoteIf* methods
-    } else if (curr->is<Switch>()) {
-      auto* sw = curr->cast<Switch>();
-      auto targets = BranchUtils::getUniqueTargets(sw);
+    } else {
+      // TODO support BrOn, Switch, etc. where possible
+      auto targets = BranchUtils::getUniqueTargets(curr);
```

Comment:
Is this currently supporting `BrOn` and `Switch` or not? The `TODO` sounds like it isn't, but then what is this code doing?

Also, is there any reason the support for `BrOn` or `Switch` should be different from that of `Break`, which is handled above, given that we are using generic methods like `getUniqueTargets`?

Reply:
I'll clarify that better, good point. Basically, the code supports BrOn and Switch, but it ignores them. The TODO is to actually optimize them.

I think you're right that we could use generic methods to handle them. We'd also need to generically check for an existing return value (as the Break code does), as well as for the number of targets (Break can assume there is just one). For now though I'd prefer to not look into those things.